pingpong/msg_pingpong.c

```diff
@@ -278,8 +278,15 @@ static int run(void)
 	}
 
 	ret = wait_for_completion(txcq, max_credits - credits);
+	if (ret)
+		goto out;
+
 	/* Finalize before closing ep */
-	ft_finalize(fi, ep, txcq, rxcq, FI_ADDR_UNSPEC);
+	if (opts.dst_addr)
+		ret = ft_client_finalize(fi, ep, txcq, FI_ADDR_UNSPEC);
+	else
+		ret = ft_server_finalize(rxcq);
+
```

Comment:
Okay, if we decide that this is a fix we want to pull then I will merge them and use the addr passed in to determine server/client.